spec-tests/src/main/java/org/candlepin/spec/bootstrap/data/util/FileUtil.java

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (c) 2009 - 2022 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.spec.bootstrap.data.util;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipInputStream;
+
+/**
+ * Utility methods for managing system files.
+ */
+public final class FileUtil {
+
+    private FileUtil() {
+        throw new UnsupportedOperationException();
+    }
+
+    /**
+     * Unzips a compressed file into a directory with the same name
+     * in the same directory. Also unzips child directories if they
+     * are zipped.
+     *
+     * @param zipPath path of the zip file.
+     * @return file of the directory containing the uncompressed zip contents.
+     * @throws IOException if unable to decompress.
+     */
+    public static File unzip(Path zipPath) throws IOException {
+        Path targetDirectory = Paths.get(zipPath.toString().replace(".zip", ""));
+        return unzip(zipPath, targetDirectory);
+    }
+
+    /**
+     * Unzips a compressed file into a target directory. Also unzips
+     * child directories if they are zipped.
+     *
+     * @param zipPath path of the zip file.
+     * @param targetDirectory location of the directory to export the contents to.
+     * @return file of the directory containing the uncompressed zip contents.
+     * @throws IOException if unable to decompress.
+     */
+    public static File unzip(Path zipPath, Path targetDirectory) throws IOException {
+        if (targetDirectory == null) {
+            throw new IllegalArgumentException("Target directory is null.");
+        }
+
+        if (!isZip(zipPath)) {
+            return null;
+        }
+
+        Files.createDirectory(targetDirectory);
+        try (ZipInputStream zipInputStream = new ZipInputStream(Files.newInputStream(zipPath))) {
```

Comment:
I've been looking at this for a bit now, and I'm beginning to wonder: given that we can query for a ZipEntry specifically off a ZipFile, is it necessary for our tests to even unzip the export archives we receive? We could just treat it as a fully-contained file system and avoid needing most of this class and the cleanup it mandates. Moreover, if we ensure the export file itself is created as a temp file (via File.createTempFile), then it'll automatically be cleaned up by the JVM and we won't need to do any management of it at all.

Reply:
Updated to use the ZipFile as a self-contained files system and ensured that it is cleaned up by the JVM! This cleaned up a lot!